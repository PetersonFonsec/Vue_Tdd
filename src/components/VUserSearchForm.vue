<template>
    <div>        
        <form @submit.prevent="$emit('submitted', username)">
            <input type="text" v-model="username" />
            <button type="submit">Enviar</button>
        </form>
    </div>
</template>

<script>
export default {
    name: 'VUserSearchForm',
    data(){
        return {
            username:''
        }
    }
}
</script>

<style>
</style>
