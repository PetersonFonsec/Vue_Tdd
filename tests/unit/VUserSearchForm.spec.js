import { shallowMount } from '@vue/test-utils'
import VUserSearchForm  from '@/components/VUserSearchForm' 

describe('VUserSearchForm', ()=>{
    const build = () => {
        const wrapper = shallowMount(VUserSearchForm)

        return {
            wrapper,
            input : ()=> wrapper.find('input'),
            button : ()=> wrapper.find('button'),
        }
    }

    it('renders the components', ()=> {
        const { wrapper } = build()

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('calls "submitted" event when submiting a form ', ()=>{
        const expectedUser = 'kuroski'
        const { wrapper, input, button } = build()

        input().element.value = expectedUser

        input().trigger('input')
        button().trigger('click')
        button().trigger('submit')

        expect( wrapper.emitted().submitted[0] ).toEqual([expectedUser])
    })
})