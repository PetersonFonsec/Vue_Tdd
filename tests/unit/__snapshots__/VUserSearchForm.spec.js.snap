// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUserSearchForm renders the components 1`] = `
<div>
  <form><input type="text"> <button type="submit">Enviar</button></form>
</div>
`;
